server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -163,6 +168,26 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a sequential unordered stream of the map entries.
+     *
+     * @return a {@link Stream} of the map entries as {@link Entry}
+     */
+    public Stream<Entry<KType, VType>> stream() {
```

Comment:
Can we use `Map.Entry` for the return value, just to make it a familiar type and to ensure that some of the helpers working on `Map.Entry` can be used too? I am not too fond of introducing yet one more key-value pair type.

Reply:
Right, I was debating which option would be better to use (internal `Entry` with public fields or `Map.Entry`). I will update the consumer code to use `Map.Entry` getters instead of field accessors!